examples/ImGuiModule.cmake

```diff
@@ -0,0 +1,312 @@
+if(NOT DEFINED ImGuiModule_CMAKE_INCLUDED)
+    set(ImGuiModule_CMAKE_INCLUDED 1)
+else()
+    return()
+endif()
+
+if(CMAKE_VERSION VERSION_LESS 3.5)
+    include(CMakeParseArguments)
+endif()
+
+if(EXISTS "${ImGui_SRCDIR}" AND EXISTS "${ImGui_SRCDIR}/imgui.cpp")
+    set(ImGui TRUE)
+else()
+    include(${CMAKE_CURRENT_LIST_DIR}/ImGuiOptions.cmake)
+endif()
```

Comment:
This is a lot of magic. If the source directory was deleted for some reason, then look for an options file? I've not seen CMake configuration like this before and it feels wildly overengineered.